spec/index.html

```diff
@@ -980,6 +980,18 @@ <h3>Examples</h3>
     <section id="canon-algo-algo">
       <h3>Algorithm</h3>
 
+      <p>The following algorithm will run with a minimal number of iterations in each step
+        for well formed <a>input datasets</a>.
+        In some extreme cases, the algorihm can behave poorly, particularly in <a href="#ca.5">Step 5</a>.
+        Implementations MUST prevent against potenial denial-of-service attacks.
+        See <a href="#dataset-poisoning"></a> for futher information.</p>
+
+      <p class="note">Implementations can consider placing limits on the number of
+        calls to <a href="#hash-nd-quads"></a> based on the number
+        of blank nodes in the <a>hash to blank nodes map</a>.
+        For most well-formed datasets, no more than a couple
+        of iterations on <a href="#hash-nd-quads"></a> per blank node would be unusual.</p>
```

Comment:
```suggestion
        For most well-formed datasets, more than a couple
        of iterations on <a href="#hash-nd-quads"></a> per blank node would be unusual.</p>
```

Note also my comment above on the term "well-formed".

(Minor: even if the term is retained, it should be consistent on whether it is "well-formed" or "well formed".)